File: frontend/src/app/models/team.ts
import { League } from './league';
import { Logo } from './logo';

export interface Stadium {
  name?: string;
  capacity?: number;
  built?: number;
  location?: string;
}

export interface Colors {
  primary?: string;
  secondary?: string;
}

export interface Achievements {
  championships?: number;
  cups?: number;
}

export interface SocialMedia {
  facebook?: string;
  twitter?: string;
  instagram?: string;
  youtube?: string;
}

export interface Team {
  id: string;
  name: string;
  fullName?: string;
  founded?: number;
  history?: string;
  website?: string;
  league?: League;
  primaryLogo?: Logo;
  historicalLogos?: Logo[];
  stadium?: Stadium;
  colors?: Colors;
  achievements?: Achievements;
  socialMedia?: SocialMedia;
  path?: string;
  tags?: string[];
}

File: backend/src/main/java/com/gfl/api/model/Team.java
package com.gfl.api.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.util.List;
import java.util.ArrayList;

@Entity
@Table(name = "team")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Team {

    @Id
    private String id;

    private String name;

    private String fullName;

    private Integer founded;

    @Column(columnDefinition = "TEXT")
    private String history;

    private String website;

    @ManyToOne
    @JoinColumn(name = "league_id")
    private League league;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "primary_logo_id")
    private TeamLogo primaryLogo;

    @OneToMany(mappedBy = "team", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<TeamLogo> historicalLogos = new ArrayList<>();

    @Embedded
    private Stadium stadium;

    @Embedded
    private Colors colors;

    @Embedded
    private Achievements achievements;

    @Embedded
    private SocialMedia socialMedia;
}
